### PR TITLE
add options for strict testing

### PR DESCRIPTION
**Pull request by pawelru**
- part of https://github.com/insightsengineering/coredev-tasks/issues/478
  - please read this for more info about the implementation: https://github.com/insightsengineering/coredev-tasks/issues/478#issuecomment-1909912778
- fixes discovered partial argument matches

### CODE REVIEW

File: R/validate_inputs.R
Comment (chlebowa) [inline R/validate_inputs.R:1]: What is the purpose of these changes?

Comment (pawelru) [reply R/validate_inputs.R:1]: In addition to strict testing I also applied strict options in examples. For examples with shiny apps, I used a simple mocking described [here](https://github.com/insightsengineering/teal.modules.clinical/pull/983#issue-2102483923).

Without this change I got the following:
```
r$> ### Name: validate_inputs
r$> ### Title: Send input validation messages to output
r$> ### Aliases: validate_inputs
r$> 
r$> ### ** Examples
r$> 
r$> library(shiny)

r$> library(shinyvalidate)

r$> ui <- fluidPage(

(...)

r$> if (interactive()) {
+   shinyApp(ui, server)
+ }
hello from mocked runApp
Error in `app_initialize()`:
! Identified global objects via static code inspection (function (input, output); {; iv <- InputValidator$new(); iv$add_rule("letter", sv_in_set(LETTERS, "choose a capital letter")); iv$add_rule("number", ~if (as.integer(.)%%2L == 1L); "choose an even number"); ...; col = input[["color"]], cex = input[["size"]]); }); }). Failed to locate global object in the relevant environments: ‘.’


ℹ You can inspect the failed AppDriver object via `rlang::last_error()$app`
ℹ AppDriver logs:
{shinytest2} R info 17:56:55.49 Start AppDriver initialization
{shinytest2} R info 17:56:55.50 Error while initializing AppDriver:
                                Identified global objects via static code inspection (function (input, output); {; iv <- InputValidator$new(); iv$add_rule("letter", sv_in_set(LETTERS, "choose a capital letter")); iv$add_rule("number", ~if (as.integer(.)%%2L == 1L); "choose an even number"); ...; col = input[["color"]], cex = input[["size"]]); }); }). Failed to locate global object in the relevant environments: ‘.’

Caused by error in `globalsByName()`:
! Identified global objects via static code inspection (function (input, output); {; iv <- InputValidator$new(); iv$add_rule("letter", sv_in_set(LETTERS, "choose a capital letter")); iv$add_rule("number", ~if (as.integer(.)%%2L == 1L); "choose an even number"); ...; col = input[["color"]], cex = input[["size"]]); }); }). Failed to locate global object in the relevant environments: ‘.’
```

I would love to extend strict testing with strict examples and for this I need to reach some conclusion regarding technology used but there are other priorities right now.

Comment (pawelru) [reply R/validate_inputs.R:1]: Generally speaking, NSE creates a lot of headache for static code analysis because it flags variables as global objects as shown above.
Within the package, R CMD CHECK [raises a note](https://community.rstudio.com/t/how-to-solve-no-visible-binding-for-global-variable-note/28887) and one of the recommendations is to use `.data$` pronoun. I simply applied that in here.

I believe this is the first time when we apply such tooling on examples and not surprisingly we discover already known errors.

Honestly speaking I dislike it too. Examples must be simple and this introduces unneeded syntax clutter. I just don't have any better idea... We can of course remove it but I bet it will circle back - I expect: soonish.

Comment (chlebowa) [reply R/validate_inputs.R:1]: We could rewrite the example to use functions instead of formulas in validation rules:
from: `iv_par$add_rule("color", ~ if (length(.) > 1L) "choose only one color")`
to: &nbsp;&nbsp;&nbsp;&nbsp;`iv_par$add_rule("color", function(value) if (length(value) > 1L) "choose only one color")`

And `plot` can be rewritten to use vectors instead of a formula:
from: `plot(eruptions ~ waiting, faithful)`
to: &nbsp;&nbsp;&nbsp;&nbsp;`plot(faithful$waiting, faithful$eruptions)`

Comment (pawelru) [reply R/validate_inputs.R:1]: Thanks! I will try this out

Comment (pawelru) [reply R/validate_inputs.R:1]: applied in the newest commit - please tell me what you think about it

Comment (chlebowa) [reply R/validate_inputs.R:1]: Looks much better, thank you :+1:
What do you think about adding braces to the rule functions for more clarity?

Comment (pawelru) [reply R/validate_inputs.R:1]: Good point. This might be more clear for some of the users. Will add that shortly.

**Review by chlebowa — APPROVED** [review]
:+1:
